packages/widgets/src/calendar/serverData.ts

```diff
@@ -0,0 +1,30 @@
+"use server";
+
+import type { RouterOutputs } from "@homarr/api";
+import { api } from "@homarr/api/server";
+
+import type { WidgetProps } from "../definition";
+
+export default async function getServerDataAsync({ integrationIds }: WidgetProps<"calendar">) {
+  try {
+    const data = await api.widget.calendar.findAllEvents({
+      integrationIds,
+      itemId: "" // TODO: Get actual item ID here
```

Comment:
You can add it in the `definition.ts` file of widgets package to the `WidgetProps` (just move the itemId from `WidgetComponentProps` up there). After that you only need to add it to the `server/runner.tsx` in widgets package